src/main/java/com/microsoft/azure/relay/ClientWebSocket.java

```diff
@@ -326,11 +328,7 @@ public void onClose(Session session, CloseReason reason) {
 		this.fragmentQueue.shutdown();
 		
 		this.closeReason = reason;
-		if (reason.getCloseCode() != CloseReason.CloseCodes.NORMAL_CLOSURE) {
-			this.closeTask.completeExceptionally(new RuntimeIOException("Websocket did not close properly: " + reason.toString()));
-		} else {
-			this.closeTask.complete(null);
-		}
```

Comment:
Having the "((LifeCycle) this.container).stop();" line above will sometimes cause the close reason to be ones that's not NORMAL_CLOSURE (such as DISCONNECT), so taking out the throw to avoid problems

Reply:
Does this result in ever hiding relevant information from the user?

---
In reply to: [260548769](https://github.com/Azure/azure-relay-java/pull/24#discussion_r260548769) [](ancestors = 260548769)

Reply:
There is the logging line above that records the close reason, so even if something unexpected happened it should be recorded

---
In reply to: [260851931](https://github.com/Azure/azure-relay-java/pull/24#discussion_r260851931) [](ancestors = 260851931,260548769)

Reply:
As a general point (I'm not sure that it applies here and am not suggesting there's further work for this PR) if we have info that users may want to use to make a decision at runtime then we should give it to them.

In this case if there's nothing they might want to do different depending on whether it was graceful close or not then it may be fine to not pass this info along.

---
In reply to: [260866112](https://github.com/Azure/azure-relay-java/pull/24#discussion_r260866112) [](ancestors = 260866112,260851931,260548769)

Reply:
Also, customers rarely gather traces from what I've observed. :(

---
In reply to: [260867220](https://github.com/Azure/azure-relay-java/pull/24#discussion_r260867220) [](ancestors = 260867220,260866112,260851931,260548769)

Reply:
Ugh, I just realized which class this is. Customers of ClientWebSocket may indeed want to know what happened at the protocol level when they requested close happen a certain way and the remote server didn't cooperate. I could see an argument that the users of ClientWebSocket might want the details if close wasn't graceful. 

However, users of the HybridConnection API are less likely to need details if they asked for a close and something happened less than gracefully. Since they don't need to act differently, like call some abort method, then they don't necessarily need to know about it.

---
In reply to: [260868130](https://github.com/Azure/azure-relay-java/pull/24#discussion_r260868130) [](ancestors = 260868130,260867220,260866112,260851931,260548769)

Reply:
What if we pass the CloseReason as the result to the closeTask? so closeAsync() returns CF<CloseReason> instead of CF<Void>?

---
In reply to: [260867220](https://github.com/Azure/azure-relay-java/pull/24#discussion_r260867220) [](ancestors = 260867220,260866112,260851931,260548769)

Reply:
I think that would be great!

---
In reply to: [260870149](https://github.com/Azure/azure-relay-java/pull/24#discussion_r260870149) [](ancestors = 260870149,260867220,260866112,260851931,260548769)

Reply:
Actually, it's already available through getCloseReason!

---
In reply to: [260871743](https://github.com/Azure/azure-relay-java/pull/24#discussion_r260871743) [](ancestors = 260871743,260870149,260867220,260866112,260851931,260548769)